app.js

```diff
@@ -301,15 +302,39 @@ function ClientOptionsHandler(fileSources) {
     };
     this.setCompilers = compilers => {
         const blacklistedKeys = ['exe', 'versionFlag', 'versionRe', 'compilerType', 'demangler', 'objdumper',
-            'postProcess'];
+            'postProcess', 'demanglerClassFile', 'isSemVer'];
+        let semverGroups = [];
         const copiedCompilers = JSON.parse(JSON.stringify(compilers));
-        _.each(options.compilers, (compiler, compilersKey) => {
+        _.each(copiedCompilers, (compiler, compilersKey) => {
+            // Find if this compiler's group uses semver
+            if (compiler.isSemVer && !semverGroups.includes(compiler.group)) {
+                semverGroups.push(compiler.group);
+            }
             _.each(compiler, (_, propKey) => {
                 if (blacklistedKeys.includes(propKey)) {
                     delete copiedCompilers[compilersKey][propKey];
                 }
             });
         });
+        // Order those groups which have semver defined
+        const groupped = _.groupBy(copiedCompilers, compiler => compiler.group);
+        _.each(semverGroups, group => {
```

Comment:
I think you can do an `_.each(grouped, group => group.sort(....)` here: `_.each()` on a map will call each method with `(value, key, list)`. Saves a `groupped[group]` access.